api/docs/http_certificate_revocation_list.md

```diff
@@ -5,7 +5,11 @@ The `certificate_revocation_list` endpoint retrieves a Certificate Revocation Li
 from the master.  The master must be configured to be a CA.  The returned
 CRL is always in the `.pem` format.
 
-In all requests the `:environment` and `:nodename` must be given, but neither has any bearing on the request.
+Under Puppet Server's CA service, the `environment` parameter is ignored and can
+be omitted. Under a Rack or WEBrick Puppet master, `environment` is required and
+must be a valid environment, but it has no effect on the response.
+
+The `:nodename` is required, but has no effect on the response.
```

Comment:
That response seemed suspicious, so I checked the default auth.conf, and [sure enough:](https://github.com/puppetlabs/puppet/blob/master/conf/auth.conf#L94) 

```
# allow all nodes to access the certificates services
path /puppet-ca/v1/certificate_revocation_list/ca
method find
allow *
```

If you were to change the auth rules, I'm positive that asking for the "foo" CRL would work identically to asking for the "ca" one.